core/components/scripts/scripts.tsx

```diff
@@ -0,0 +1,121 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import Script from 'next/script';
+
+interface ScriptNode {
+  entityId: string;
+  consentCategory: string;
+  location: string;
+  visibility: string;
+  __typename?: 'InlineScript' | 'SrcScript';
+  scriptTag?: string;
+  src?: string;
+  integrityHashes: Array<{
+    hash: string;
+  }>;
+}
+
+interface ScriptsData {
+  edges: Array<{
+    node: ScriptNode;
+  }> | null;
+}
+
+interface ScriptRendererProps {
+  scripts: ScriptsData | null;
+  location: 'head' | 'footer';
+  strategy: 'afterInteractive' | 'lazyOnload';
+}
+
+interface ScriptsProps {
+  scripts: ScriptsData | null;
+}
+
+interface ScriptProps {
+  src?: string;
+  dangerouslySetInnerHTML?: {
+    __html: string;
+  };
+  strategy: 'afterInteractive' | 'lazyOnload';
+  integrity?: string;
+  id?: string;
+  crossorigin?: string;
+}
```

Comment:
A lot of these props can be computed via the `FragmentOf` util exported from the client + Next.js `Script` props. Can we update this to utilize those instead?